lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -580,6 +581,38 @@ struct RvalueExprVisitor {
     return conditionalOp.getResult();
   }
 
+  /// Handle calls.
+  Value visit(const slang::ast::CallExpression &expr) {
+    // Class method calls are currently not supported.
+    if (expr.thisClass()) {
+      mlir::emitError(loc, "unsupported class method call");
+      return {};
+    }
+    return std::visit(
+        [&](auto &subroutine) { return visitCall(expr, subroutine); },
+        expr.subroutine);
+  }
+
+  /// Handle subroutine calls.
+  Value visitCall(const slang::ast::CallExpression &expr,
+                  const slang::ast::SubroutineSymbol *subroutine) {
+    mlir::emitError(loc, "unsupported subroutine call");
+    return {};
+  }
+
+  /// Handle system calls.
+  Value visitCall(const slang::ast::CallExpression &expr,
+                  const slang::ast::CallExpression::SystemCallInfo &info) {
+    const auto &subroutine = *info.subroutine;
+
+    if (subroutine.name == "$signed" || subroutine.name == "$unsigned")
+      return context.convertRvalueExpression(*expr.arguments()[0]);
+
+    mlir::emitError(loc) << "unsupported system call `" << subroutine.name
+                         << "`";
+    return {};
```

Comment:
I viewed the source code in slang. I understood that `SubroutineSymbol*` is used to handle the functions defined by users like `function ... endfunction`, but the `SystemCallInfo` handles the system functions like `$signed`, `$display`, etc, right?

Reply:
Yes exactly! 🙂

Reply:
Oh! I learn it! :smile: